src/async_lcd.rs

```diff
@@ -2,49 +2,48 @@ use embedded_hal_async::{delay::DelayNs, i2c::I2c};
 
 use crate::{
     Backlight, BitMode, Commands, CursorMoveDir, DisplayControl, DisplayShift, Font, Mode,
+    OFFSETS_16X4, OFFSETS_NORMAL,
 };
 
 /// API to write to the LCD.
-pub struct Lcd<'a, I, D>
+pub struct Lcd<'a, const ROWS: u8, const COLUMNS: u8, I, D>
 where
     I: I2c,
     D: DelayNs,
 {
     i2c: &'a mut I,
     address: u8,
-    rows: u8,
     delay: &'a mut D,
     backlight_state: Backlight,
     cursor_on: bool,
     cursor_blink: bool,
     font_mode: Font,
 }
 
-impl<'a, I, D> Lcd<'a, I, D>
+impl<'a, const ROWS: u8, const COLUMNS: u8, I, D> Lcd<'a, ROWS, COLUMNS, I, D>
 where
     I: I2c,
     D: DelayNs,
 {
     /// Create new instance with only the I2C and delay instance.
     pub fn new(i2c: &'a mut I, delay: &'a mut D) -> Self {
+        assert!(ROWS > 0, "ROWS needs to be larger than zero!");
```

Comment:
THere might be a clever way to compile-time-check these constants, but I don't know it from the top of my head. I'll keep it in mind to investigate

Reply:
I was hoping the asserts might do it, but sadly they don't and I didn't find another way. But it could very well be that there is something that I missed